src/vs/workbench/parts/files/electron-browser/files.contribution.ts

```diff
@@ -298,6 +298,12 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('formatOnSave', "Format a file on save. A formatter must be available, the file must not be auto-saved, and editor must not be shutting down."),
 			'overridable': true,
 			'scope': ConfigurationScope.RESOURCE
+		},
+		'editor.formatOnSaveTimeout': {
```

Comment:
Setting is editor-global, it will be hard for end user to figure out correct timeout as it depends on the language and the machine speed.

Reply:
True. Made it overridable, then at least every extension that identifies this as a potential issue can provide a sane default (as well as the user).